mlos_core/mlos_core/optimizers/bayesian_optimizers/smac_optimizer.py

```diff
@@ -103,15 +131,14 @@ def __init__(self, *,  # pylint: disable=too-many-locals,too-many-arguments
         self._temp_output_directory: Optional[TemporaryDirectory] = None
 
         # pylint: disable=import-outside-toplevel
-        from smac import HyperparameterOptimizationFacade as Optimizer_Smac
         from smac import Scenario
-        from smac.intensifier.abstract_intensifier import AbstractIntensifier
         from smac.main.config_selector import ConfigSelector
         from smac.random_design.probability_design import ProbabilityRandomDesign
-        from smac.runhistory import TrialInfo
 
         # Store for TrialInfo instances returned by .ask()
-        self.trial_info_map: Dict[ConfigSpace.Configuration, TrialInfo] = {}
+        self.trial_info_df: pd.DataFrame = pd.DataFrame(
+            columns=["Configuration", "Metadata", "TrialInfo", "TrialValue"]
+        )
```

Comment:
Why replacing this with a dataframe?

Reply:
Because configurations are no longer a unique identifier if we support metadata.

An alternative would be to use the type:

`Dict[Tuple[ConfigSpace.Configuration, pd.DataFrame], Tuple[TrialInfo, TrialValue]]`

but I thought using a dataframe was simpler and more flexible

Reply:
Not certain you can actually use that Tuple as a Dict key (may not be hashable).

Reply:
Tuples are hashable, but dataframes are not